### ISSUE TITLE
torch.onnx.export error

### ISSUE THREAD

**Issue by xiaohuzi1996** [pytorch/pytorch#105526]
### 🐛 Describe the bug

while running the code

```python
input_s = torch.zeros((1, 500), dtype=torch.long).to('cuda')
input_names = ["x"] # specify the name of the input tensor
output_names = ["out"] # specify the name of the output tensor
dynamic_axes = {"x":{0: "batch_size"}, "out":{0: "batch_size"}}
torch.onnx.export(model, input_s, 'model_test.onnx', input_names=input_names, output_names=output_names)
```

The following problem occurs, what is the situation, is this a common warning, or an error? do i need to edit again

### Versions

 UserWarning: The exported ONNX model failed ONNX shape inference.The model will not be executable by the ONNX Runtime.If this is unintended and you believe there is a bug,please report an issue at https://github.com/pytorch/pytorch/issues.Error reported by strict ONNX shape inference: [ShapeInferenceError] Shape inference error(s): (op_type:MaxPool, node name: /MaxPool): [ShapeInferenceError] Attribute strides has incorrect size
(op_type:MaxPool, node name: /MaxPool_1): [ShapeInferenceError] Attribute strides has incorrect size
(op_type:MaxPool, node name: /MaxPool_2): [ShapeInferenceError] Attribute strides has incorrect size
 (Triggered internally at ../torch/csrc/jit/serialization/export.cpp:1407.)
  _C._check_onnx_proto(proto)
============= Diagnostic Run torch.onnx.export version 2.0.1+cu117 =============
verbose: False, log level: Level.ERROR
======================= 0 NONE 0 NOTE 0 WARNING 0 ERROR ========================

**Comment by xiaohuzi1996** [comment]
But the model_test.onnx model file is generated

**Comment by MaanavD** [comment]
Hi @xiaohuzi1996 

We’ve gone ahead and closed this issue because we think it can be resolved with our new `torch.onnx.dynamo_export` api. Please try that instead! Apologies for not getting to this in time.

**Comment by fxmarty** [comment]
@MaanavD `torch.onnx.dynamo_export` does not offer the same features as `torch.onnx.export` so I don't think closing all issues related to torchscript + ONNX is the right way to go.

**Comment by thiagocrepaldi** [comment]
@fxmarty pytorch project doesnt use the wontfix label. Issues are closed with a message to let the author know (one of) the reason(s). 

Op has reported a shape inference error being reported, but without an actionable repro script, we can't help. Given `torch.onnx.export` is not as actively maintained as `torch.onnx.dynamo_export`, we recommended the latter instead of the former.

Both `torch.onnx.dynamo_export` and `torch.onnx.export` have the same ultimate goal of providing correct ONNX models, but they are not meant to provide the same export options. This is not possible because Dynamo and TorchScript (the engines of both exporters) are very different technologies with different capabilities, limitations and approaches to trace graphs.
